pkg/status/cel-evaluator.go

```diff
@@ -48,7 +54,8 @@ func newCELEvaluator() (*celEvaluator, error) {
 		cel.Declarations(
 			decls.NewVar(sourceObjectKey, decls.NewMapType(decls.String, decls.Dyn)),
 			decls.NewVar(returnedKey, decls.NewMapType(decls.String, decls.Dyn)),
-			decls.NewVar(inventoryKey, decls.NewMapType(decls.String, decls.String)),
+			decls.NewVar(inventoryKey, decls.NewMapType(decls.String, decls.String)), // contains name:string only
+			decls.NewVar(propagationMetaKey, decls.NewMapType(decls.String, decls.Dyn)),
```

Comment:
It is possible, and preferable, to declare the specific keys in this map and the type of each one's associated value.